test/path.jl

```diff
@@ -85,6 +85,29 @@
     end
     @test relpath(S(joinpath("foo","bar")), S("foo")) == "bar"
 
+    @testset "splitpath" begin
+        @test ["a", "b", "c"] == splitpath(joinpath("a","b","c"))
+        @test [] == splitpath("")
+
+        if Sys.iswindows()
+            @test ["C:", "a", "b", "c"] == splitpath("C:\\\\a\\b\\c")
+            @test ["C:"] == splitpath("C:\\\\")
+            @test ["J:"] == splitpath("J:\\\\")
+            @test ["a"] == splitpath("a\\")
+            @test ["a","b"] == splitpath("a\\\\b\\\\")
+            # TODO: what should this do for a leading slash?
+            # (what does dirname() do for a leading slash?)
+            # @test ["\\", "a","b"] == splitpath("\\a\\b\\")
+        else
+            @test ["/", "a", "b"] == splitpath("/a/b")
+            @test ["/"] == splitpath("/")
+            @test ["a"] == splitpath("a/")
+            @test ["a", "b"] == splitpath("a/b/")
```

Comment:
Meep. I guess you're probably right:
```julia
julia> splitdir("a/")
>> ("a", "")
julia> splitdir("a///")
>> ("a", "")
```

I have never liked when utilities treat paths a with trailing slash differently, because tools seem to be inconsistent with when they add one. Haha oh, I guess I just described Postel's Law! Learned something new today. But yeah, like with Postel's Law, I think I wanted to be flexible with the input, so it would work whether or not there was a trailing slash. And I can't imagine why someone would want an empty path component in an array of the path components...

\*But\* I also think consistency across these functions is important, so I'm totally happy to go with whatever you think is best.

Reply:
Fwiw, python3 does not add the trailing `""`:

```python3
>>> from pathlib import PurePosixPath, PureWindowsPath
>>> PurePosixPath('/path/to/file.txt').parts
('/', 'path', 'to', 'file.txt')
>>>
>>> PurePosixPath('/path/to/dir/').parts
('/', 'path', 'to', 'dir')
```

Reply:
Let's follow Python here—that's what I did for the rest of the path utils. I also think it's the right call, having thought about it some more: otherwise code that calls `splitpath` always has to look out for the trailing slash corner case. What was pushing me in the other direction was wanting to have some kind of identity that `joinpath(splitpath(path)) == path`, but I think avoiding the trailing slash gotcha is more important. That is probably why the Python function works the way it does.

Reply:
ah yeah that's too bad that `joinpath(splitpath(path))` can change the trailing slash....

I was going to say that actually `joinpath(splitpath(path)) == normpath(path)`, and it almost does, except for that it always trims trailing slashes. But I think that's okay, for all the reasons we just cited above. :) Thanks!